### PR TITLE
use environment hook

### PR DESCRIPTION
**Pull request by v1v**
this can help consumers to use environment hooks accessing the google cloud. 

For instance https://github.com/avaly/gcp-secret-manager-buildkite-plugin

```yaml

steps:
  - command: 'echo \$SECRET_VAR'
    plugins:
      - elastic/oblt-google-auth#v1.1.0:
          lifetime: 7200 # 2 hours
      - avaly/gcp-secret-manager#v1.1.1:
          credentials_file: /etc/gcloud-credentials.json
          env:
            SECRET_VAR: my-secret-name
            OTHER_SECRET_VAR: my-other-secret-name
```

### CODE REVIEW

File: hooks/environment [hunk 1]
```diff
@@ -32,3 +32,4 @@ JSON
 export BUILDKITE_OIDC_TMPDIR=$TMPDIR
 export GOOGLE_APPLICATION_CREDENTIALS=$TMPDIR/credentials.json
 export CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=$GOOGLE_APPLICATION_CREDENTIALS
+export BUILDKITE_OIDC_AUDIENCE=$AUDIENCE
```
Comment (v1v) [inline hooks/environment]: can you explain this?

Comment (reakaleek) [reply hooks/environment]: Sorry, I was just debugging.

Comment (reakaleek) [reply hooks/environment]: https://github.com/elastic/oblt-google-auth-buildkite-plugin/pull/2/commits/c74bd404efef2579000fd0434ea9fe622373bf17 this works now.

it's inspired by https://github.com/google-github-actions/auth/blob/main/src/main.ts#L213

**Review by reakaleek — APPROVED** [review]

File: hooks/environment [hunk 2]
```diff
@@ -29,6 +30,11 @@ cat << JSON > "$TMPDIR"/credentials.json
 }
 JSON
 
-export BUILDKITE_OIDC_TMPDIR=$TMPDIR
-export GOOGLE_APPLICATION_CREDENTIALS=$TMPDIR/credentials.json
-export CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=$GOOGLE_APPLICATION_CREDENTIALS
+export BUILDKITE_OIDC_TMPDIR="${TMPDIR}"
```
Comment (v1v) [inline hooks/environment:33]: What's the reason for this variable?

Comment (reakaleek) [reply hooks/environment:33]: This is used in the [pre-exit](https://github.com/elastic/oblt-google-auth-buildkite-plugin/blob/130a9797010e836a2ba1260938471d4300e116fb/hooks/pre-exit#L5) hook.

**Review by reakaleek — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**